File: S2P2_herencia/P2A4Empleado/P2A4C++/EmpleadoAsalariado.cpp
#include "EmpleadoAsalariado.h"
#include <sstream>
#include <iomanip>

EmpleadoAsalariado::EmpleadoAsalariado(const std::string& nombre, const std::string& apellido, 
                                       const std::string& nss, double salario)
    : Empleado(nombre, apellido, nss) {
    establecerSalarioSemanal(salario);
}

void EmpleadoAsalariado::establecerSalarioSemanal(double salario) {
    salarioSemanal = (salario < 0.0) ? 0.0 : salario;
}

double EmpleadoAsalariado::obtenerSalarioSemanal() const {
    return salarioSemanal;
}

double EmpleadoAsalariado::ingresos() const {
    return obtenerSalarioSemanal();
}

std::string EmpleadoAsalariado::toString() const {
    std::ostringstream output;
    output << std::fixed << std::setprecision(2);
    output << "empleado asalariado: " << Empleado::toString()
           << "\nsalario semanal: $" << obtenerSalarioSemanal();
    return output.str();
}

File: 1POO/Semestre 1 - 2/S2P2_herencia/P2A4Empleado/P2A4Python/EmpleadoAsalariado.py
from Empleado import Empleado

class EmpleadoAsalariado(Empleado):
    def __init__(self, nombre, apellido, nss, salario):
        super().__init__(nombre, apellido, nss)
        self._salarioSemanal = 0.0
        self.establecerSalarioSemanal(salario)

    def establecerSalarioSemanal(self, salario):
        self._salarioSemanal = 0.0 if salario < 0.0 else salario

    def obtenerSalarioSemanal(self):
        return self._salarioSemanal

    def Ingresos(self):
        return self.obtenerSalarioSemanal()

    def __str__(self):
        return f"empleado asalariado: {super().__str__()}\nsalario semanal: ${self.obtenerSalarioSemanal():,.2f}"
